dbms/src/Flash/Mpp/ExchangeReceiver.cpp

```diff
@@ -544,42 +539,14 @@ void ExchangeReceiverBase<RPCContext>::reactor(const std::vector<Request> & asyn
 
     while (alive_async_connections > 0)
     {
-        // to avoid waiting_for_retry_requests starvation, so the max continuously popping
-        // won't exceed 100ms (check_waiting_requests_freq * timeout) too much.
-        static constexpr Int32 check_waiting_requests_freq = 10;
-        static constexpr auto timeout = std::chrono::milliseconds(10);
+        AsyncHandler * handler = nullptr;
+        ready_requests.pop(handler);
```

Comment:
why not check the result of `ready_requests.pop(handler)` ?

Reply:
> It seems the result of `ready_requests.pop(handler)` is not checked, is it on purpose?

Yes, it's unnecessary to check the success of pop as we can ensure that it will always be successful.